modules/nf-core/propr/propr/main.nf

```diff
@@ -8,18 +8,23 @@ process PROPR_PROPR {
         'biocontainers/r-propr:5.0.3' }"
 
     input:
-    tuple val(meta), path(count)
+    tuple val(meta),
+    path(count),
+    path(opt_file)
 
     output:
     tuple val(meta), path("*.propr.rds"), emit: propr
     tuple val(meta), path("*.propr.tsv"), emit: matrix
     tuple val(meta), path("*.fdr.tsv"),   emit: fdr         , optional:true
+    tuple val(meta), path("*.gct"),	  emit: gct	    , optional:true
+    tuple val(meta), path("*.cls"),	  emit: cls	    , optional:true
     path "*.R_sessionInfo.log",           emit: session_info
     path "versions.yml",                  emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
+    sample_file = (opt_file) ? "$opt_file" : 'EMPTY'
```

Comment:
what is the sample_file for? If it only sets a condition, I still recommend you to have it only in 'ext.args' and not here in path(opt_file).
Also you should always only parse the optional arguments inside the template.

Reply:
I have removed it, it was a remnant of generating the CLS file and I was also basically using at a flag for GSEA output generation. Since the script does not generate CLS anymore it does not need a sample sheet so now both the propr.R and propd.R templates have a GSEA flag. I have removed any optional input from the main.nf file